components/search_results/index.tsx

```diff
@@ -48,24 +55,65 @@ function makeMapStateToProps() {
                 if (channel && channel.delete_at !== 0 && !viewArchivedChannels) {
                     return;
                 }
+                channels[post.channel_id] = channel;
 
                 posts.push(post);
             });
         }
 
+        const newFilesResults = getSearchFilesResults(state);
```

Comment:
I'm worried about React reconciliations here given that we'll be creating a new object each time as a prop, forcing this component to re-render for /every/ state change.

Can we explore managing these more discretely via reducers instead?

Reply:
This is based on the pre-existing code related to posts. I think is not needed to change it as part of this PR.